resolver/resolver.go

```diff
@@ -407,7 +407,37 @@ func isUDP(w dns.ResponseWriter) bool {
 // permitted length of the given transmission channel.
 // See https://tools.ietf.org/html/rfc1035#section-4.2.1
 func truncate(m *dns.Msg, udp bool) *dns.Msg {
-	m.Truncated = udp && m.Len() > dns.MinMsgSize
+	bufsize := 0
+	switch udp {
+	case true:
+		if opt := m.IsEdns0(); opt != nil {
+			bufsize = int(opt.UDPSize())
+		}
+		if bufsize < dns.MinMsgSize {
+			bufsize = dns.MinMsgSize
+		}
+	case false:
+		bufsize = dns.MaxMsgSize - 1
```

Comment:
[ Quoting <notifications@github.com> in "Re: [mesos-dns] Remove answers to f..." ]
>> +	case false:
>> +		bufsize = dns.MaxMsgSize - 1
>
>@miekg I was referring to @tsenart's comment:
>
>> Why the - 1?
>
>I saw this in your code here:
>https://github.com/skynetservices/skydns/blob/4c00898b5ed8769af24f8dde381386f3969c2275/server/server.go#L205

I think that as because dns.MaxMsgSize was once 2^16, now it is defined as
2^16-1, so the -1 can be removed here.

>Beside that, this code here sets the truncated bit when removing. But in the meanwhile I changed it based on @tsenart suggestions to make it more readable.

Ack.

/Miek

--
Miek Gieben